### ISSUE TITLE
cpp, zsdcpp:  pre-processors do not treat system include path differently from user include path

### ISSUE THREAD

**Issue by aralbrec** [z88dk/z88dk#316]
The search path is no different for files included within angle brackets and quotes:
```
#include <stdio.h>

void main(void)
{
   return;
}
```

If you create a stdio.h in the current directory and compile, the local file will be included instead of the system file.

If you don't have a local stdio.h but instead do a `#include "stdio.h"` the system include file will be found.

This bug is occurring in #313 where a user-supplied input.h is present which is being included in preference to the system input.h when `#include <input.h>` is used.

**Comment by suborb** [comment]
I suspect sdcpp supports -isystem. Zcpp certainly won't. 

So if you change the appropriate -I over hopefully the problem will go away for sdcc

**Comment by aralbrec** [comment]
Hmm, I've put -isystem in but zsdcpp is still preferring the local header.

**Comment by suborb** [comment]
Maybe a space after the option?

I looked at zcpp earlier and it goes through the directories in order, presumably the order on the command line. 

I'll have a play later though.

**Comment by aralbrec** [comment]
I tried space too but it's still searching the local dir first.  But I did find this:
```
The most commonly-used option is -Idir, which causes dir to be searched after the current directory (for the quote form of the directive) and ahead of the standard system directories.
```

which means -I trumps the system search path.

It looks like -iquote and -isystem is what is needed instead of -I.  I'll put those in as equivalent to -I for zcpp.

**Comment by suborb** [comment]
It looks like there's a few issues coming into play in my tests:

* The .cfg file OPTIONS line has -I. in it
* zcc pushes user specified -I ahead of the system path
* zcpp doesn't have the distinction between quote and system path

**Comment by aralbrec** [comment]
Yes the cfg files have to be changed to use -iquote and -isystem, zcc has to quote the filenames when it sees those options and zcpp has to have iquote and isystem undestandable as -I.

zcpp is doomed to get this wrong until it is updated to meaningfully use iquote and isystem.  The -I behaviour is correct though - it does get applied ahead of isystem.

**Comment by suborb** [comment]
I think the above implements the feature for zcpp, it's a bit ugly, but lets see if it works.

I've found a few #include problems in the classic library as result as well.

**Comment by aralbrec** [comment]
I've made the necessary changes to zcc and I made -iquote and -isystem synonyms for -I in zcpp so things would continue to work as normal.  The changes are simple so any conflict should be easy to fix with your zcpp additions.

I made changes to zx.cfg too to use -iquote and -isystem and it seems to work for zsdcpp/zsdcc.

The rest of the cfg files will have to be changed similarly but let's see if it also works with zcpp when you've committed that.
